cli/src/pcluster/api/controllers/cluster_compute_fleet_controller.py

```diff
@@ -49,6 +60,37 @@ def update_compute_fleet_status(cluster_name, update_compute_fleet_status_reques
 
     :rtype: None
     """
-    update_compute_fleet_status_request_content = UpdateComputeFleetStatusRequestContent.from_dict(
-        update_compute_fleet_status_request_content
-    )
+    try:
+        update_compute_fleet_status_request_content = UpdateComputeFleetStatusRequestContent.from_dict(
+            update_compute_fleet_status_request_content
+        )
+        cluster = Cluster(cluster_name)
+        if not check_cluster_version(cluster, exact_match=True):
+            raise BadRequestException(
+                f"the update can be performed only with the same ParallelCluster version ({cluster.stack.version}) "
+                "used to create the cluster."
+            )
+        status = update_compute_fleet_status_request_content.status
+        if cluster.stack.scheduler == "slurm":
+            if status == RequestedComputeFleetStatus.START_REQUESTED:
+                cluster.start()
+            elif status == RequestedComputeFleetStatus.STOP_REQUESTED:
+                cluster.stop()
```

Comment:
I think we need to rework the error handling for start and stop operations cause they are not handled correctly with the current version of the patch. We should probably use a more granular exception in the model rather than ClusterActionError as done for the other APIs (e.g. create) so that the error is properly handled upstream in the controllers. Let's make sure these errors conditions are covered by unit tests.

Reply:
I only make a granular exception if the stack is in `IN_PROGRESS` status. I will check tomorrow if there are other granular exceptions I can use.